src/components/RangeSelector.js

```diff
@@ -130,20 +172,30 @@ class RangeSelector extends React.Component {
       newState[this.state.dragging.toLowerCase() + 'Pixels'] = newPixels;
       newState[this.state.dragging.toLowerCase() + 'Value'] = newValue;
 
-      this.setState(newState);
+      this.setState(newState, () => {
+        this.props['on' + this.state.dragging + 'DragStop'](this.state[this.state.dragging.toLowerCase() + 'Value']);
+
+        if (this.state.trackClicked) {
+          this.setState({
+            dragging: false,
+            trackClicked: false
+          });
+        }
+      });
```

Comment:
oh nice - I was thinking that since `setState()` is async (right?) the callback would be necessary. Thanks for the feedback I'll fix this today.